lib/Sema/CSRanking.cpp

```diff
@@ -1142,6 +1142,17 @@ SolutionCompareResult ConstraintSystem::compareSolutions(
         ++score1;
       continue;
     }
+
+    // If one is a function type and the other is a key path, prefer the other.
+    if (type1->is<FunctionType>() != type2->is<FunctionType>()) {
+      auto anyKeyPathTy = cs.DC->mapTypeIntoContext(
+          cs.DC->getASTContext().getAnyKeyPathDecl()->getDeclaredType());
```

Comment:
mapTypeIntoContext() is a no-op here since getDeclaredType() always returns unbound generic types.

Reply:
Thanks for the tip. Is there a better way to recognize key paths than this?